### ISSUE TITLE
Default magmom behavior causes magnetic ordering to reset between relax -> static

### ISSUE THREAD

**Issue by mattmcdermott** [materialsproject/atomate2#505]
The default handling of magmoms (#375) results in an unexpected behavior that causes magnetic orderings to be reset to FM in between VASP calculations in a flow.

Consider a flow consisting of a relax followed by a static (`RelaxMaker` -> `StaticMaker`) for an input structure with an AFM magnetic ordering. After the relaxation, when the structure is copied and the input set is being generated for the static calculation, one of two cases can happen for each species in the structure: 

- A) the magmom is replaced if its value is defined in the input set  (e.g., `BaseVaspSet.yaml`). This is always a positive value (no negative magmoms, hence the AFM ordering is destroyed).
- B) the magmom is copied from the previous calculation's output structure. This happens for non-magnetic species.

This annotated snippet of code from [_get_magmoms()](https://github.com/materialsproject/atomate2/blob/0978f5de2c16cdb1746d2f1f8faaad6b8a4f22e2/src/atomate2/vasp/sets/base.py#L872) shows why:

```python
for site in structure:
    specie = str(site.specie)
    if specie in magmoms:
        mag.append(magmoms.get(specie))   # Case A: magnetic species get replaced from input set value
    elif hasattr(site, "magmom"):
        mag.append(site.magmom)  # Case B: non-magnetic species get copied
    elif hasattr(site.specie, "spin") and site.specie.spin is not None:
        mag.append(site.specie.spin)
```

The ordering of these statements leads to AFM orderings being destroyed between calculations.

@utf @janosh -- is this behavior an unintended consequence?

**Comment by utf** [comment]
Yes, this is the intended behaviour in that it is consistent with the pymatgen input sets. E.g., see here:

https://github.com/materialsproject/pymatgen/blob/a5b82b5a2c6dd66fef6bfea87f39d3b6656770cb/pymatgen/io/vasp/sets.py#L456-L460

**Comment by utf** [comment]
Actually, sorry I see your point. Yes, this looks like it is wrong.

**Comment by utf** [comment]
Actually, this takes a bit more thinking about. As far as I'm aware, the pymatgen static set default is to NOT copy the magmoms from the previous structure. E.g., see this code in the `MPStaticSet` which will override the magmoms from the previous `INCAR` (not the structure): https://github.com/materialsproject/pymatgen/blob/a5b82b5a2c6dd66fef6bfea87f39d3b6656770cb/pymatgen/io/vasp/sets.py#L1112-L1118

Now in our case, the order of preference should be:

  1. user incar settings
  2. magmoms in input struct
  3. spins in input struct
  4. job config dict
  5. set all magmoms to 0.6

However, as you suggest, there is a bug in that the magmoms from the structure will never get set if that species is defined in the config dict. I think the issue is really here: 

https://github.com/materialsproject/atomate2/blob/0978f5de2c16cdb1746d2f1f8faaad6b8a4f22e2/src/atomate2/vasp/sets/base.py#L629-L631

where `v` is the either the `user_incar_settings` magmoms if they have been set or the config dict magmoms if not. Instead, I think v should always just be the `user_incar_settings` magmoms. E.g., the code should be changed to:

```python
 incar[k] = _get_magmoms( 
     structure, config_magmoms=config_magmoms, magmoms=user_incar_settings.get("MAGMOMS", {})
 ) 
```

I'd appreciate you double checking this logic.

**Comment by mattmcdermott** [comment]
Hi @utf, I appreciate the quick response. I tested out your suggested fix and it works :) 

This logic makes sense to me based on the order of preference.

I can make a PR for this. Should we add a test too?

**Comment by utf** [comment]
Thanks for checking. Yeah, it would be great if you could submit a PR with a fix and a test for the various precedences!

**Comment by mattmcdermott** [comment]
addressed in #506 !
